### DISCUSSION TITLE
Workspace/Default SDK

### DISCUSSION THREAD

**Discussion by joaoLouceiro** [nvim-java/nvim-java#180] [Q&A]
I'm pretty new to vim/neovim and this is the first time I'm actually trying to use this as my IDE.

Problem: when I run :JavaRunnerRunMain from inside my project's folder, it's trying to use Java_21:
![image](https://github.com/nvim-java/nvim-java/assets/89097431/b869836a-6df7-476f-b200-e0ba5d712690)

This is asking me to run with '--enable-preview' which, honestly, I have no idea how to do in this case, so if you can help with that, I'd appreciate it.

Weird thing, for me, is that, if I run this exact same class from the parent folder, it used a different SDK (which should work without --enable-preview"):
![image](https://github.com/nvim-java/nvim-java/assets/89097431/81a052b8-a057-412c-b003-42319ff8ece8)

Never mind all the "was compiled by x, you're trying to run with y". 

Where can I find where nvim-java looks for the SDK?

**Comment by s1n7ax** [comment]
@joaoLouceiro I don't know a way to pass compiler arguments to `jdtls`.
This is how you can configure the SDK
https://github.com/nvim-java/nvim-java?tab=readme-ov-file#clamp-how-to-use-jdk-xx-version

**Comment by joaoLouceiro** [comment]
Hey, guys!

So, I think I (kinda) got it figured out.

Basically, the JDTLS workspace configurations are stored inside the ~/.cache/nvim/jdtls/workspaces
I guess my workspace just stuck to the SDK version that was set when I opened it the first time.

Now, since I'm a newborn in this thing, and this was my first project, I could deal with simply deleting the whole workspace, reopening nvim, trying to run the :JavaRunnerRunMain command and, ta-da, it's pointing to my default SDK.
There's probably some way to change which SDK it's looking at, but I couldn't really bother to look for it.

For now, this seems to have solved the problem.

If not, it's a step in the right direction and I hope it helps someone.

Cheers!
